### ISSUE TITLE
Express req.ip is returning ::ffff:127.0.0.1

### ISSUE THREAD

**Issue by Alex0007** [indutny/node-ip#31]
Based on [SO question](http://stackoverflow.com/questions/29411551/express-js-req-ip-is-returning-ffff127-0-0-1)

How to parse it?

**Comment by bruce-one** [comment]
I'd suggest checking out https://github.com/whitequark/ipaddr.js

    _.isEqual(ipaddr.process('::ffff:127.0.0.1'), ipaddr.process('127.0.0.1'))

**Comment by bryanwayb** [comment]
I had the same issue, went ahead an forked the repo, fixed, and waiting on a merge.

You can view the pull request here: https://github.com/indutny/node-ip/pull/35

**Comment by yinrong** [comment]
ip-1.0.1 still have problem with transition ipv4 address

ip.isPrivate("::ffff:127.0.0.1") return false

**Comment by bryanwayb** [comment]
Hmm, weird, I'd not taken a look at that method before, had just assumed it converted the passed IP address and verified from there.

Seems that method is using regular expressions to do validation, which is fine for IPv4 but IPv6 is a bit more volatile in parsing. Basically it works, but only for each variation that get's manually added.

Try this: `npm install https://github.com/bryanwayb/node-ip.git`

Also, my PR had been closed, but there's a few reasons to open a new one. Will reference it here once I've done so.

**Comment by bryanwayb** [comment]
New pull request with the fix is #38

**Comment by bryanwayb** [comment]
This has since been fixed in version 1.0.2.
